### ISSUE TITLE
Parens Matching

### ISSUE THREAD

**Issue by johnjelinek** [vsClojure/vsClojure#92]
It would be great if vsClojure provided Parens Matching like Emacs does.

Example: If I do an open paren, it automatically creates a closing paren and unless I go into "paren-edit" mode, I cannot delete the closing parens.

**Comment by frankhale** [comment]
A lot of people hate paredit but I actually like it and this is definitely a good feature request to have!

**Comment by frankhale** [comment]
Might as well through in rainbow parenthesis as well.

**Comment by johnjelinek** [comment]
That would be AWESOME

**Comment by frankhale** [comment]
Parens are matched, see screenshot:

![paren-matching](https://f.cloud.github.com/assets/190154/1745442/fcacf814-642d-11e3-8cf0-aa74b5033c88.PNG)

**Comment by speige** [comment]
I think John was asking for it to auto-create the closing paren when you type the opening paren. This could be useful, but I don't see it as critical.

**Comment by johnjelinek** [comment]
Yea, at least having rainbow parens would be a step in the right direction.

**Comment by frankhale** [comment]
Good point @speige, something like paredit would be nice. Fortunately there a lot of examples out there in the Clojure world that implement this so if we need we can use those as reference material. The Eclipse Counterclockwise Clojure plugin has a paredit module as well as Emacs and a few others that I know of off the top of my head.
